usr/lib/cca_stdll/cca_specific.c

```diff
@@ -447,6 +447,57 @@ static CK_BBOOL analyse_cca_key_token(const CK_BYTE *t, CK_ULONG tlen,
     return FALSE;
 }
 
+static CK_RV check_expected_mkvp(STDLL_TokData_t *tokdata,
+                                 enum cca_token_type keytype,
+                                 const CK_BYTE *mkvp)
+{
+    struct cca_private_data *cca_private = tokdata->private_data;
+    const char *mktype;
+    const CK_BYTE *expected_mkvp;
+
+    switch (keytype) {
+    case sec_des_data_key:
+        expected_mkvp = cca_private->expected_sym_mkvp;
+        mktype = "SYM";
+        break;
+
+    case sec_aes_data_key:
+    case sec_aes_cipher_key:
+    case sec_hmac_key:
+        expected_mkvp = cca_private->expected_aes_mkvp;
+        mktype = "AES";
+        break;
+
+    case sec_rsa_priv_key:
+    case sec_ecc_priv_key:
+        expected_mkvp = cca_private->expected_apka_mkvp;
+        mktype = "APKA";
+        break;
```

Comment:
Shouldn't be RSA = ASYM, but ECC = APKA?

Reply:
No APKA is used for 'newer' asymmetric key tokens, and this includes besides ECC also the newer RSA key token types that opencryptoki is using.

Reply:
Ok, if you are sure ... Well the 7.1 CCA docu says:
 On the cryptographic coprocessor, PKA keys are protected by the Asymmetric-Keys Master Key (ASYM-MK).
The ASYM-MK is a triple-length DES key used to protect PKA private keys. On the cryptographic coprocessor, the ASYM-MK protects RSA private keys. There are two PKA master keys: the ASYM-MK mentioned above, and the 256-bit AES PKA Master Key (APKA-MK), used to protect ECC private keys stored in ECC key tokens.

Reply:
The CCA doc is not very consistent when it talks about APKA master keys and where they are used. They always mention ECC keys, but only sometimes mention newer RSA key tokens. I guess the confusion is because the APKA master keys were introduced with CCA's ECC support, and thus ECC were the first that uses APKA keys. When newer RSA key tokens were added later on, they forgot to adjust many places in the manual that talk about APKA and ECC only....

See my comment below that hopefully clarifies it.